frontend/src/pages/Report.tsx

```diff
@@ -240,6 +267,51 @@ export const Report = () => {
     return;
   };
 
+  const findRowHours = (rowTopic: IssueActivityPair, days: Date[]) => {
+    let rowHours = [];
+    days.map((day) => {
+      let hours = 0;
+      let entry: TimeEntry | FetchedTimeEntry = newTimeEntries?.find(
+        (entry) =>
+          entry.spent_on === formatDate(day, "yyyy-MM-dd") &&
+          entry.issue_id === rowTopic.issue.id &&
+          entry.activity_id === rowTopic.activity.id
+      );
+      if (!entry && timeEntries && timeEntries.length > 0) {
+        entry = timeEntries?.find(
+          (entry) =>
+            entry.spent_on === formatDate(day, "yyyy-MM-dd") &&
+            entry.issue.id === rowTopic.issue.id &&
+            entry.activity.id === rowTopic.activity.id
+        );
+      }
+      if (entry) {
+        hours = entry.hours;
+      }
+      rowHours.push(hours);
+      return;
+    });
+    return rowHours;
+  };
+
+  const findRowEntryIds = (rowTopic: IssueActivityPair, days: Date[]) => {
+    let rowEntryIds = [];
+    days.map((day) => {
+      let id = 0;
+      let entry = timeEntries?.find(
+        (entry) =>
+          entry.spent_on === formatDate(day, "yyyy-MM-dd") &&
+          entry.issue.id === rowTopic.issue.id &&
+          entry.activity.id === rowTopic.activity.id
+      );
+      if (entry) {
+        id = entry.id;
+      }
+      rowEntryIds.push(id);
+    });
+    return rowEntryIds;
+  };
+
```

Comment:
These are two new functions AFAIK: I think they could each have a separate commit.

Reply:
They were moved from Row and only slightly modified, that's why I put it into one commit. But I agree. Will try to think of this in the future.